test/conformance/suite_test.go

```diff
@@ -95,9 +100,13 @@ func TestMain(m *testing.M) {
 	// let's ensure that conformance related namespaced are deleted from the cluster.
 	exitOnErr(ensureConformanceTestsNamespacesAreNotPresent(ctx, env.Cluster().Client()))
 
-	code := m.Run()
-
-	os.Exit(code)
+	code = m.Run()
+	if testenv.KeepTestCluster() == "" && testenv.ExistingClusterName() == "" {
+		ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+		defer cancel()
+		fmt.Printf("INFO: cluster %s is being deleted\n", env.Cluster().Name())
+		exitOnErr(env.Cleanup(ctx))
+	}
```

Comment:
We already have an implementation for this in https://github.com/Kong/kubernetes-ingress-controller/blob/a3875a8944ec579e7e0f886760a7d70e5f34401f/test/internal/helpers/teardown.go#L25-L31.

While this does accept `*testing.T` instead of `*testing.M` we could refactor this a bit so that we can share the implementation details and so that have a second function which can be called without `*testing.T` and reuse it here and in https://github.com/Kong/kubernetes-ingress-controller/blob/07b2023e1e20f051e239dc9691934c3b40c93eb6/test/integration/suite_test.go#L236-L241.

WDYT?